core/log/profiler_hook_summary.cpp

```diff
@@ -155,38 +191,39 @@ struct summary {
             entries.back().name = name;
         }
         const auto id = it->second;
-        stack.emplace_back(id, now);
-        overhead_ns +=
-            std::chrono::duration_cast<std::chrono::nanoseconds, int64>(
-                clock::now() - now)
-                .count();
+        auto now = get_current_time_point();
+        stack.emplace_back(id, std::move(now));
+        overhead += cpu_clock::now() - cpu_now;
     }
 
     void pop(const char* name, bool allow_pop_root = false)
     {
-        const auto now = clock::now();
+        const auto cpu_now = cpu_clock::now();
         std::lock_guard<std::mutex> guard{mutex};
+        auto now = get_current_time_point();
         if (!check_pop_status(*this, name, allow_pop_root)) {
             return;
         }
         const auto id = stack.back().first;
-        const auto partial_entry = stack.back();
+        auto partial_entry = std::move(stack.back());
         stack.pop_back();
         auto& entry = entries[id];
-        const auto elapsed_ns =
-            std::chrono::duration_cast<std::chrono::nanoseconds, int64>(
-                now - partial_entry.second)
-                .count();
+        const auto cpu_now2 = cpu_clock::now();
+        // we need to exclude the wait for the timer from the overhead
+        // measurement
+        timer->wait(now);
```

Comment:
should pop also contain the wait when the profilehook contains synchronize feature?
although I think the event logger is broken if the device does not execute them yet.
another concern: 
should pop get time when it pops? pop only pops the time_point out for the start/end pair but do not compute the time.
you will have a time_point list and it will be filled by the time information by execution.
only compute the time difference when writing the summary.
We can have only one sync before writing the summary not each pop

Reply:
yes, I also thought it also require more work to do fully asynchronized timing.
I mean, if we do not enable the synchronization and do not have wait() here, the time event will possibly be not set yet to lead broken.

Reply:
I see, then I understood you correctly :)